spec/javascript/alchemy_admin/components/button.spec.js

```diff
@@ -0,0 +1,64 @@
+import "alchemy_admin/components/button"
+import { renderComponent } from "./component.helper"
+
+describe("alchemy-button", () => {
+  it("disables button on form submit", () => {
+    const html = `
+      <form>
+        <button type="submit" is="alchemy-button">Save</button>
+      </form>
+    `
+    const button = renderComponent("alchemy-button", html)
+    const submit = new Event("submit", { bubbles: true })
+
+    button.form.dispatchEvent(submit)
+
+    expect(button.getAttribute("disabled")).toEqual("disabled")
+    expect(button.getAttribute("tabindex")).toEqual("-1")
+    expect(button.classList.contains("disabled")).toBeTruthy()
+    expect(button.innerHTML).toEqual(
+      '&nbsp;<alchemy-spinner size="small" color="currentColor"></alchemy-spinner>'
+    )
+  })
+
+  it("logs warning if no form found", () => {
+    global.console = {
+      ...console,
+      warn: jest.fn()
+    }
+
+    const html = `
+    <button is="alchemy-button">Save</button>
+    `
+    const button = renderComponent("alchemy-button", html)
+
+    expect(console.warn).toHaveBeenCalledWith(
+      "No form for button found!",
+      button
+    )
+  })
+
+  describe("on remote forms", () => {
+    it("re-enables button on ajax complete", () => {
+      const html = `
+        <form data-remote="true">
+          <button type="submit" is="alchemy-button">Save</button>
+        </form>
+      `
+      const button = renderComponent("alchemy-button", html)
+
+      const submit = new Event("submit", { bubbles: true })
+      button.form.dispatchEvent(submit)
+
+      expect(button.getAttribute("disabled")).toEqual("disabled")
```

Comment:
You could split that into different describe and it - functions. Jest is pretty fast it would desribe, what are you going to test.